integration/spark/src/main/scala/org/apache/spark/sql/CarbonCountStar.scala

```diff
@@ -108,4 +125,173 @@ case class CarbonCountStar(
     CarbonInputFormatUtil.setDataMapJobIfConfigured(job.getConfiguration)
     (job, carbonInputFormat)
   }
+
+  // The detail of query flow as following for pure partition count star:
+  // Step 1. check whether it is pure partition count star by filter
+  // Step 2. read tablestatus to get all valid segments, remove the segment file cache of invalid
+  // segment and expired segment
+  // Step 3. use multi-thread to read segment files which not in cache and cache index files list
+  // of each segment into memory. If its index files already exist in cache, not required to
+  // read again.
+  // Step 4. use multi-thread to prune segment and partition to get pruned index file list, which
+  // can prune most index files and reduce the files num.
+  // Step 5. read the count from pruned index file directly and cache it, get from cache if exist
+  // in the index_file <-> rowCount map.
+  private def getRowCountPurePartitionPrune: Long = {
+    var rowCount: Long = 0
+    val prunedPartitionPaths = new java.util.ArrayList[String]()
+    val names = relation.catalogTable match {
```

Comment:
Looks like same code(Get the current partitions from table) is present in CarbonLateDecodeStrategy.pruneFilterProject. Can reuse it

Reply:
@Indhumathi27 val tableSegmentIndexes = DataMapStoreManager.getInstance().getAllSegmentIndexes(
          carbonTable.getTableId)
        if (!tableSegmentIndexes.isEmpty) {
          // clear invalid cache
          for (segmentFilePathInCache <- tableSegmentIndexes.keySet().asScala) {
            if (!validSegmentPaths.contains(segmentFilePathInCache)) {
              // means invalid cache
              tableSegmentIndexes.remove(segmentFilePathInCache)
            }
          }
        }
        // init and put absent the valid cache
        for (validSegmentPath <- validSegmentPaths) {
          if (tableSegmentIndexes.get(validSegmentPath) == null) {
            val segmentIndexMeta = new SegmentIndexMeta(validSegmentPath)
            tableSegmentIndexes.put(validSegmentPath, segmentIndexMeta)
          }
        }

like this in code, we will get that object by table id, and put info into it.

Reply:
we are adding info to local variable `tableSegmentIndexes` only right. Not to DataMapStoreManager.segmentIndexes

Reply:
@Indhumathi27 ok, will reuse the code of partition prune logic